accessibility_api/accessibility_lib/utils/WinUtil.py

```diff
@@ -0,0 +1,199 @@
+
+        if accptr not in WinUtil.simple_elements:
+            WinUtil.simple_elements[accptr] = [childid]
+        else:
+            WinUtil.simple_elements[accptr].append(childid)
+
+    @staticmethod
+    def _traverse(node, visited, search_criteria):
+        """
+        Traverse through accessible tree looking for node with the given ID
+        """
+
+        if WinUtil.match_criteria(node, search_criteria):
+            WinUtil.target = node
+            WinUtil.target.isSimpleElement = False
+            return
+
+        if DEBUG_ENABLED:
+            print_accessible(node)
+
+        # Retrieve simple children or accessible children from node
+        acc_children = WinUtil._accessible_children(node)
+
+        # Traverse through simple elements of node
+        if node in WinUtil.simple_elements:
+            for childid in WinUtil.simple_elements[node]:
+                if DEBUG_ENABLED:
+                    print_simple(node, childid)
+
+                if WinUtil.match_criteria(node, search_criteria, childid):
+                    WinUtil.target = node
+                    WinUtil.target.isSimpleElement = True
+                    WinUtil.target.childId = childid
+                    return
+
+        # Traverse through accessible objects of node
+        for child in acc_children:
+            if child not in visited:
+                visited.add(node)
+                WinUtil._traverse(child, visited, search_criteria)
+
+    @staticmethod
+    def get_indentifiers(params):
+        """
+        Get accessibility indentifiers
+        """
+
+        identifiers = {
+            'Name': params.get('name'),
+            'Role': params.get('role')
+        }
+        filtered_identifiers = {k: v for k, v in identifiers.items() if v}
+
+        return filtered_identifiers
+
+    @staticmethod
+    def match_criteria(node, params, child_id=CHILDID_SELF):
+        """
+        Find matching accessible with given criteria
+        """
+
+        search_criteria = WinUtil.get_indentifiers(params)
+        for criteria in search_criteria:
+            prefix = 'acc'
+            prop_value = getattr(node, prefix + criteria)(child_id)
+            search_value = search_criteria[criteria]
+
+            # If value is a number convert from unicode to int
+            if isinstance(prop_value, int):
+                search_value = int(search_value)
+
+            if prop_value != search_value:
+                return False
+
+        return True
+
+    @staticmethod
+    def _get_test_window():
+        """
+        Get test window handle
+        """
+
+        # Get the window for browser
+        test_class = c_char_p('MozillaWindowClass')
+        current_hwnd = windll.user32.FindWindowA(test_class, None)
+        name = (
+            WinUtil._accessible_object_from_window(current_hwnd)
+            .accName(CHILDID_SELF)
+        )
+        # Iterate through windows
+        while name is None or 'Marionette Accessible' not in name:
```

Comment:
This stuff needs to be parametrized and taken out all the way up so it could be configured at the point of creating a server or something like that. I think it's going to be hard to guarantee names in general